packages/playwright-core/src/server/bidi/bidiPage.ts

```diff
@@ -180,6 +180,12 @@ export class BidiPage implements PageDelegate {
     const frameId = params.context;
     this._page._frameManager.frameRequestedNavigation(frameId, params.navigation!);
 
+    // url is missing from navigationStarted events on Firefox when the
+    // navigation is interrupted by a beforeunload prompt.
+    // See https://bugzilla.mozilla.org/show_bug.cgi?id=1908952
+    if (!params.url)
```

Comment:
This would just mask the problem and potentially other issues elsewhere. Let's revert this and keep the test failing (it stops timing out with the other fix in this PR, right?). Our primary objective is to ensure the logic functions correctly when the protocol implementation provides the property, rather than working around the current issues in the Bidi protocol.